### PR TITLE
FIX : HTML in ODT templates

### PR DESCRIPTION
**Pull request by atm-adrien**
## FIX : HTML in ODT templates

There is currently no interpretation of HTML in ODT type document templates.

Here is what we have in entry : 
`<p><strong><u>DESCRIPTION DES PRODUITS</u></strong><br />
Ceci est un produit &amp; un service<br />
Contr&ocirc;le des stock de ce produit&nbsp;<br />
V&eacute;rification des dispositifs d&#39;installation pour ce produit&nbsp;<br />
Vidange des filtres &agrave; carburant suppl&eacute;mentaire pour ce produit&nbsp;<br />
N&deg; de ce produit</p>`

We have the same thing written in our ODT document

### CODE REVIEW

File: htdocs/includes/odtphp/odf.php [hunk 1]
```diff
@@ -161,7 +161,8 @@ public function setVars($key, $value, $encode = true, $charset = 'ISO-8859')
 	 */
 	public function convertVarToOdf($value, $encode = true, $charset = 'ISO-8859')
 	{
-		$value = $encode ? htmlspecialchars($value) : $value;
+        $value = dol_html_entity_decode($value, ENT_COMPAT);
```
Comment (eldy) [inline htdocs/includes/odtphp/odf.php:164]: Can you confirm this:  This means the parameter $encode will not be used anymore ?

**Comment by eldy** [comment]
It seems a different fix were done in v18.
Can you check to backport done into the v18 instead of this version for v17 ?

**Comment by atm-adrien** [comment]
I copied what was done in V18, this does not seem to pose a compatibility problem

I also modified the regexp a little (it did not target all of the tags)

File: htdocs/includes/odtphp/odf.php [hunk 2]
```diff
@@ -18,980 +18,1003 @@ class OdfException extends Exception
...
+                $handlein = popen($command, 'r');
+                while (!feof($handlein)) {
+                    $read = fgets($handlein);
+                    fwrite($handle, $read);
+                    $output_arr[]=$read;
+                }
+                pclose($handlein);
+                fclose($handle);
+            }
+            dolChmod($outputfile);
```
Comment (eldy) [inline htdocs/includes/odtphp/odf.php:906]: With v17, this line of code will hang. Keep v17 code as this has no relation with the bugfix.

Comment (atm-adrien) [reply htdocs/includes/odtphp/odf.php]: Done

**Pull request merged**